src/Avalonia.Controls/Flyouts/PopupFlyoutBase.cs

```diff
@@ -201,6 +201,7 @@ protected virtual bool HideCore(bool canCancel = true)
             {
                 Target.DetachedFromVisualTree -= PlacementTarget_DetachedFromVisualTree;
                 Target.KeyUp -= OnPlacementTargetOrPopupKeyUp;
+                Target = null;
```

Comment:
To match WinUI behavior, Target should be set to null after OnClosed event callback.

Reply:
Makes sense, moved the line.